modules/storages/app/views/storages/admin/storages/edit.html.erb

```diff
@@ -1,76 +1,91 @@
+  <% else %>
+    <%= render partial: 'storages/admin/storages/one_drive/edit', locals: { f: f } %>
+  <% end %>
   <%= styled_button_tag t(:button_save), class: "-highlight -with-icon icon-checkmark" %>
 <% end %>
 
-<% if @object.oauth_application %>
+<% if @storage.oauth_application %>
   <%= render(AttributeGroups::AttributeGroupComponent.new) do |component| %>
     <% component.with_header(title: "OpenProject #{t(:'storages.label_oauth_application_details')}") %>
 
     <% component.with_attributes([
-      { key: "OpenProject #{Storages::Admin::LABEL_OAUTH_CLIENT_ID}", value: @object.oauth_application.uid },
-      { key: "OpenProject #{Storages::Admin::LABEL_OAUTH_CLIENT_SECRET}", value: "●●●●●●●●●●●●●●●●" },
-    ]) %>
+                                   { key: "OpenProject #{Storages::Admin::LABEL_OAUTH_CLIENT_ID}", value: @storage.oauth_application.uid },
+                                   { key: "OpenProject #{Storages::Admin::LABEL_OAUTH_CLIENT_SECRET}", value: "●●●●●●●●●●●●●●●●" },
+                                 ]) %>
 
     <%= link_to(t("storages.buttons.replace_openproject_oauth"),
-                replace_oauth_application_admin_settings_storage_path(@object),
+                replace_oauth_application_admin_settings_storage_path(@storage),
                 method: :delete,
                 data: { confirm: t(:'storages.confirm_replace_oauth_application') },
                 class: 'button -with-icon icon-reload') %>
   <% end %>
 <% end %>
 
-<%= render(AttributeGroups::AttributeGroupComponent.new) do |component| %>
-  <% component.with_header(title: "#{t("storages.provider_types.#{@object.short_provider_type}.name")} #{t(:'storages.label_oauth_client_details')}") %>
-
-  <% if @object.oauth_client %>
-    <% component.with_attributes([
-      { key: "#{t("storages.provider_types.#{@object.short_provider_type}.name")} #{Storages::Admin::LABEL_OAUTH_CLIENT_ID}",
-        value: @object.oauth_client.client_id },
-      { key: "#{t("storages.provider_types.#{@object.short_provider_type}.name")} #{Storages::Admin::LABEL_OAUTH_CLIENT_SECRET}",
-        value: short_secret(@object.oauth_client.client_secret) },
-    ]) %>
-
-    <%= link_to(t("storages.buttons.replace_provider_type_oauth", provider_type: t("storages.provider_types.#{@object.short_provider_type}.name")),
-                new_admin_settings_storage_oauth_client_path(@object),
-                data: { confirm: t(:'storages.confirm_replace_oauth_client') },
-                class: 'button -with-icon icon-reload') %>
-  <% else %>
-    <%= link_to(t("js.label_create"), new_admin_settings_storage_oauth_client_path(@object), class: 'button -with-icon icon-add') %>
-  <% end %>
+<% if @storage.provider_type_nextcloud? %>
+  <%= render partial: 'storages/admin/storages/nextcloud/oauth_client_section' %>
+<% elsif @storage.provider_type_one_drive? %>
+  <%= render partial: 'storages/admin/storages/one_drive/oauth_client_section' %>
 <% end %>
```

Comment:
What about deriving the partial name from the provider type?
```suggestion
 <%= render partial: "storages/admin/storages/#{@storage.short_provider_type}/oauth_client_section" %>
```

Reply:
good hint!

Reply:
done